ethereum/eth/src/main/java/org/hyperledger/besu/ethereum/eth/manager/task/RetryingGetBlockFromPeersTask.java

```diff
@@ -15,84 +15,134 @@
 package org.hyperledger.besu.ethereum.eth.manager.task;
 
 import org.hyperledger.besu.datatypes.Hash;
+import org.hyperledger.besu.ethereum.ProtocolContext;
 import org.hyperledger.besu.ethereum.core.Block;
 import org.hyperledger.besu.ethereum.eth.manager.EthContext;
 import org.hyperledger.besu.ethereum.eth.manager.EthPeer;
 import org.hyperledger.besu.ethereum.eth.manager.exceptions.IncompleteResultsException;
+import org.hyperledger.besu.ethereum.eth.manager.exceptions.NoAvailablePeersException;
 import org.hyperledger.besu.ethereum.mainnet.ProtocolSchedule;
 import org.hyperledger.besu.plugin.services.MetricsSystem;
 
+import java.util.HashSet;
 import java.util.Objects;
 import java.util.Optional;
+import java.util.Set;
 import java.util.concurrent.CompletableFuture;
-import java.util.stream.Collectors;
 
 import org.slf4j.Logger;
 import org.slf4j.LoggerFactory;
 
 public class RetryingGetBlockFromPeersTask
     extends AbstractRetryingPeerTask<AbstractPeerTask.PeerTaskResult<Block>> {
 
-  private static final int DEFAULT_MAX_RETRIES = 5;
+  private static final int DEFAULT_RETRIES = 25;
```

Comment:
is it better to check what is the max number of peers ? `--max-peers`

Reply:
make sense, since max peers config is not available there, I need to pass it before I can use it, done in the next commint

Reply:
the propagation of the maxPeers info is done in [this other PR](https://github.com/hyperledger/besu/pull/3353), it is also cherry-picked here to set the number of retries to the maxPeers value